test/unit/WrapperRegistry.ts

```diff
@@ -0,0 +1,394 @@
+import { ethers } from 'hardhat'
+import { BigNumber, Contract, Signer } from 'ethers'
+import { expect } from 'chai'
+
+interface TestContext {
+  accounts: Signer[]
+  deployer: Signer
+  userA: Signer
+  wrapperRegistry: Contract
+  mockBTC: Contract
+  mockAmpl: Contract
+  buttonToken: Contract
+  buttonToken2: Contract
+  unbuttonToken: Contract
+}
+
+describe('WrapperRegistry', () => {
+  const setupTestContext = async (): Promise<TestContext> => {
+    const accounts: Signer[] = await ethers.getSigners()
+    const [deployer, userA] = accounts
+
+    const wrapperRegistryFactory = await ethers.getContractFactory(
+      'WrapperRegistry',
+    )
+    const wrapperRegistry = await wrapperRegistryFactory
+      .connect(deployer)
+      .deploy()
+
+    const erc20Factory = await ethers.getContractFactory('MockERC20')
+    const mockBTC = await erc20Factory
+      .connect(deployer)
+      .deploy('Wood Bitcoin', 'WOOD-BTC')
+    const oracleFactory = await ethers.getContractFactory('MockOracle')
+    const mockOracle = await oracleFactory.connect(deployer).deploy()
+    await mockOracle.setData('1000000000000', true);
+    const buttonTokenFactory = await ethers.getContractFactory('ButtonToken')
+    const buttonToken = await buttonTokenFactory.connect(deployer).deploy()
+    buttonToken.initialize(mockBTC.address, 'Button Bitcoin', 'BTN-BTC', mockOracle.address)
+    const buttonToken2 = await buttonTokenFactory.connect(deployer).deploy()
+    buttonToken2.initialize(mockBTC.address, 'Button Bitcoin2', 'BTN-BTC2', mockOracle.address)
+
+    const rebasingErc20Factory = await ethers.getContractFactory(
+      'MockRebasingERC20',
+    )
+    const mockAmpl = await rebasingErc20Factory.deploy(
+      'Ampleforth',
+      'AMPL',
+      10000,
+      10000,
+    )
+
+    const unbuttonTokenFactory = await ethers.getContractFactory('UnbuttonToken')
+    const unbuttonToken = await unbuttonTokenFactory.deploy()
+
+    const initialDeposit = await unbuttonToken.INITIAL_DEPOSIT()
+    const initialRate = '1000000'
+    await mockAmpl.mint(await deployer.getAddress(), initialDeposit)
+    await mockAmpl.approve(unbuttonToken.address, initialDeposit)
+    await unbuttonToken.initialize(mockAmpl.address, 'Unbutton Ampleforth', 'UBTN-AMPL', initialRate)
+
+    return {
+      accounts,
+      deployer,
+      userA,
+      wrapperRegistry,
+      mockBTC,
+      mockAmpl,
+      buttonToken,
+      buttonToken2,
+      unbuttonToken,
+    }
+  }
+
+  describe('Initialization', function () {
+    it('Can successfully deploy BondConfigVault with proper arguments', async () => {
+      const { wrapperRegistry } = await setupTestContext()
```

Comment:
You can use fixtures https://ethereum-waffle.readthedocs.io/en/latest/fixtures.html to optimize set up time over multiple tests